PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiverAsyncClientCheck.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.NotNull;
+
+
+/**
+ * This class extends the LocalInspectionTool to check for the use of ServiceBusReceiverAsyncClient
+ * in the code and suggests using ServiceBusProcessorClient instead.
+ */
+public class ServiceBusReceiverAsyncClientCheck extends LocalInspectionTool {
+
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+
+        return new JavaElementVisitor() {
+
+            @Override
+            public void visitLocalVariable(PsiLocalVariable variable) {
+                super.visitLocalVariable(variable);
+
+                final PsiType type = variable.getType();
+                final String qualifiedName = type.getCanonicalText();
+
+                if (qualifiedName.equals("com.azure.messaging.servicebus.ServiceBusReceiverAsyncClient")) {
```

Comment:
It's not fundamentally clear whether or not the rule is only applicable to `ServiceBusReceiverAsyncClient` or if it should also apply to the synchronous counterpart. In general, this could be addressed by:

1. Plug-in documentation for the rule to assist the users
2. PR documentation to assist code-reviewers.

Adding some more information and/or documentation to both justify and scope your PR could help move forward the review process (especially when the reviewer is not as familiar with what you've been working on).

Reply:
got it:) added more comments and javadocs